OOInteraction/src/commands/CSceneHandlerItemTest.cpp

```diff
@@ -53,76 +56,133 @@ bool CSceneHandlerItemTest::canInterpret(Visualization::Item*, Visualization::It
 }
 
 Interaction::CommandResult* CSceneHandlerItemTest::execute(Visualization::Item*, Visualization::Item*,
-		const QStringList&, const std::unique_ptr<Visualization::Cursor>&)
+		const QStringList&, const std::unique_ptr<Visualization::Cursor>&) __attribute__ ((optnone))
 {
 	//Test code goes here
 
-	QString projectsDir = "projects/";
+	QString projectName = "DiffTest";
+	QString projectsDir = "projects/" + projectName;
+
+	// get GitRepository
+	if (!FilePersistence::GitRepository::repositoryExists(projectsDir))
+	{
+		qDebug() << "no repository found at " + projectsDir;
+		return new Interaction::CommandResult{};
```

Comment:
It doesn't really matter for the test command, but in general there are ways to return errors from commands.